File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_cardbill.java
// EXPECTED: 69.44
// Reference emit for CARDBILL — fee + late IF + interest pipeline.
public final class CardbillRef {
  public static void main(String[] args) {
    double bal = 1000.00;
    double feeRate = 0.0290;
    double intRate = 0.0150;
    int daysLate = 45;
    double lateFee = 25.00;
    double fee = Math.round(bal * feeRate * 100.0) / 100.0;
    double late = daysLate > 30 ? lateFee : 0.0;
    double interest = Math.round((bal + fee) * intRate * 100.0) / 100.0;
    double total = Math.round((fee + late + interest) * 100.0) / 100.0;
    System.out.printf("%.2f%n", total);
  }
}

File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_cardbill.py
"""Reference emit for CARDBILL — fee + late IF + interest → 69.44."""
# EXPECTED: 69.44
BAL = 1000.00
FEE_RATE = 0.0290
INT_RATE = 0.0150
DAYS_LATE = 45
LATE_FEE = 25.00
FEE = round(BAL * FEE_RATE + 1e-12, 2)
LATE = LATE_FEE if DAYS_LATE > 30 else 0.0
INTEREST = round((BAL + FEE) * INT_RATE + 1e-12, 2)
TOTAL = round(FEE + LATE + INTEREST + 1e-12, 2)
print(f"{TOTAL:.2f}")
